bitnami/mongodb/values.yaml

```diff
@@ -45,255 +45,319 @@ image:
 ##
 # fullnameOverride:
 
-serviceAccount:
-  # Specifies whether a ServiceAccount should be created
-  create: true
-  # The name of the ServiceAccount to use.
-  # If not set and create is true, a name is generated using the fullname template
-  name:
+## Kubernetes Cluster Domain
+##
+clusterDomain: cluster.local
 
-## Init containers parameters:
-## volumePermissions: Change the owner and group of the persistent volume mountpoint to runAsUser:fsGroup values from the securityContext section.
+## Use an alternate scheduler, e.g. "stork".
+## ref: https://kubernetes.io/docs/tasks/administer-cluster/configure-multiple-schedulers/
 ##
-volumePermissions:
-  enabled: false
-  image:
-    registry: docker.io
-    repository: bitnami/minideb
-    tag: buster
-    pullPolicy: Always
-    ## Optionally specify an array of imagePullSecrets.
-    ## Secrets must be manually created in the namespace.
-    ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
-    ##
-    # pullSecrets:
-    #   - myRegistryKeySecretName
-  resources: {}
+# schedulerName:
+
+## MongoDB architecture. Allowed values: standalone or replicaset
+##
+architecture: standalone
```

Comment:
I have a small concern, and it is that we're encapsulating the concept of replicaset inside an "architecture" value. I'm thinking of a user who would just run "helm install mongodb bitnami/mongodb".

Why is it we can't keep the previous "replicaset.enable" variable? For me it is a lot more intuitive. If we need to decide between an architecture or another, we could always create a helper for deciding the architecture depending on "replicaset.enable".

Reply:
There are two reasons why I prefer an "architecture" parameter, similar to the one we already use in MinIO (name 'mode') or InfluxDB:

- If, in the future, you want to add a 3rd supported architecture setup, you can still use the same parameter.
- It helps the user to understand that enabling replicaset has several implications. It's not like adding a feature but switching the whole architecture.